apps/create-vesting/src/components/Checkboxes.tsx

```diff
@@ -1,67 +1,145 @@
+      <div className="flex items-center">
+        <div className="relative p-1">
+          <Input
+            type="checkbox"
+            id={name}
+            {...register(name)}
+            className="sr-only" // This hides the input visually but keeps it accessible
+            {...props}
+          />
+          <label
+            htmlFor={name}
+            style={{
+              width: `${radius}px`,
+              height: `${radius}px`,
+            }}
+            className={`
+            flex items-center justify-center rounded-full border-none cursor-pointer
+            bg-color-primary-lighter hover:bg-color-primary [&>span>span]:hover:bg-color-primary
+          `}
+          >
+            <span
+              // It's weird, but tailwind classes don't work for this
+              style={{
+                width: `${(radius * 3) / 4}px`,
+                height: `${(radius * 3) / 4}px`,
+              }}
+              className={`
+              flex items-center justify-center rounded-full bg-color-paper
+              transition-opacity duration-200 ease-in-out
+            `}
+            >
+              <span
+                style={{
+                  width: `${(radius * 3) / 8}px`,
+                  height: `${(radius * 3) / 8}px`,
+                }}
+                className={`
+              block rounded-full bg-color-primary-lighter
+              transition-opacity duration-200 ease-in-out
+              ${isSelected ? "opacity-100" : "opacity-0"}
+            `}
+              />
+            </span>
+          </label>
+        </div>
+        <Label>{label}</Label>
+      </div>
+      {isSelected && isSelectedMessage && (
+        <span className="flex items-center ml-4 pt-1 font-normal text-xs text-mono opacity-70">
+          <ExclamationTriangleIcon className="w-4 h-4 mr-1" />
+          {isSelectedMessage}
+        </span>
+      )}
     </div>
   );
 };
 
+export const VestUserInputCheckbox = () => {
+  const { control, setValue } = useFormContext();
+  const vestUserInput = useWatch({ control, name: "vestUserInput" });
+  const vestAllFromSwap = useWatch({ control, name: "vestAllFromSwap" });
+  const vestAllFromAccount = useWatch({ control, name: "vestAllFromAccount" });
+
+  return (
+    <Checkbox
+      name="vestUserInput"
+      label="Input vesting amount manually"
+      unselectTrigger={vestAllFromAccount || vestAllFromSwap}
+      onSelectSideEffect={() => setValue("amount", "")}
+      disabled={vestUserInput}
+    />
+  );
+};
+
 export const VestAllFromSwapCheckbox = () => {
-  const { vestAllFromSwap, setVestAllFromSwap } = useTokenAmountTypeContext();
+  const { control, setValue } = useFormContext();
+  const vestUserInput = useWatch({ control, name: "vestUserInput" });
+  const vestAllFromSwap = useWatch({ control, name: "vestAllFromSwap" });
+  const vestAllFromAccount = useWatch({ control, name: "vestAllFromAccount" });
```

Comment:
I think that if you implement the `onChange` logic on the `Checkbox` component you don't need these watches anymore. 

In addition, the `disabled` property could be created inside the checkbox component (since you're watching the same variable there).

Reply:
nice, the components got way cleaner with this